src/wp-includes/class-wp-query.php

```diff
@@ -3054,6 +3059,55 @@ public function get_posts() {
 		 */
 		$this->posts = apply_filters_ref_array( 'posts_pre_query', array( null, &$this ) );
 
+		$cache = false;
+		if ( $q['post_query_cache'] ) {
+			$cache_args = $q;
+
+			unset(
+				$cache_args['suppress_filters'],
+				$cache_args['cache_results'],
+				$cache_args['fields'],
+				$cache_args['update_post_meta_cache'],
+				$cache_args['update_post_term_cache'],
+				$cache_args['lazy_load_term_meta']
+			);
+			$key          = md5( serialize( $cache_args ) );
```

Comment:
The cache key of the other query classes include the SQL query too, this one probably should too https://github.com/WordPress/wordpress-develop/blob/f3441e37eddc5f98ffbe2c51ceebf800797e6d7f/src/wp-includes/class-wp-term-query.php#L785

I think it's mainly a safety in case a developer uses the DB clause filters directly and the args and SQL don't match.

Reply:
Fixed.